### PR TITLE
fix: more scalable code and registry

### PR DESCRIPTION
**Pull request by berteotti**
- Remove contracts with failing `getTradableOrder()`
- Only call CoW API with validTo in the future

Current version:
image
New version:
image

### CODE REVIEW

File: actions/watch.ts [hunk 1]
```diff
@@ -26,15 +29,35 @@ export const checkForAndPlaceOrder: ActionFn = async (
...
         [Array.from(order)]
       );
-      console.log(`Placing Order: ${order}`);
-      await placeOrder(
-        { ...order, from: contract_address, signature },
-        chainContext.api_url
-      );
-    } catch (e) {
+
+      if (order.validTo * 1000 < new Date().getTime()) {
```
Comment (Diogomartf) [inline actions/watch.ts]: Suggestion: I think we could turn this into a variable that describes the condition.

`const orderIsInvalid = order.validTo * 1000 < new Date().getTime()`

eg.
```suggestion
      if (orderIsInvalid) {
```

Comment (Diogomartf) [reply actions/watch.ts]: I actually like it more when the first condition is valid.

```js
if(orderIsValid)
  valid logic
else
   invalid logic
```

File: actions/watch.ts [hunk 2]
```diff
@@ -26,15 +29,35 @@ export const checkForAndPlaceOrder: ActionFn = async (
...
+  for (const contract_address of contractsToDelete) {
+    const contractIndex = registry.contracts.findIndex((existing: string) => existing == contract_address);
+    if (contractIndex >= 0) {
+      registry.contracts.splice(contractIndex, 1);
+      console.log(`Removing contract ${contract_address}`);
+    }
+  }
+  if(contractsToDelete.length > 0) {
+    registry.write();
+  }
```
Comment (Diogomartf) [inline actions/watch.ts:61]: shouldn't we also reset the `contractToDelete`?

Comment (berteotti) [reply actions/watch.ts:61]: No, `contractToDelete` doesn't persist between new action executions

File: actions/watch.ts (hunk 1 above)
Comment (berteotti) [reply actions/watch.ts]: fixed

**Review by Diogomartf — APPROVED** [review]

**Review by ElRodrigote — APPROVED** [review]

**Pull request merged**